internal/app_config.go

```diff
@@ -97,40 +100,45 @@ func (c AppConfig) GetDefault(key string, defval string) string {
 	return val
 }
 
-func (c AppConfig) Write(keyVals map[string]string) error {
-	// It's unlikely but you could theoretically have keyvals in more than
-	// one config file. This makes its hard to do atomically. So, we should
-	// find the most significant config file and put all the keyvals into
-	// that one.
-	// We also have to assert the file we are writing isn't the z-50-fioctl.toml
-	// file. This file is "managed" so doing a config write to a keyval in that
-	// file would never work - it would get overwritten.
-	mostSignificatIdx := 0
-	found := false
+// findWritableFile looks at the keyVals and determines which toml file
+// we should update.
+//
+// It's unlikely but you could theoretically have keyvals in more than
+// one config file. This makes its hard to do atomically. So, we should
+// find the most significant config file and put all the keyvals into
+// that one.
+//
+// Returns nil if no writable file is found
+func (c AppConfig) findWritableFile(keyVals map[string]string) (*cfgFile, error) {
 	for i := range c.cfgs {
 		for k := range keyVals {
 			if c.cfgs[i].tree.Has(k) {
-				found = true
-				mostSignificatIdx = i
-				break
+				// We also have to assert the file we are writing isn't the z-50-fioctl.toml
+				// file. This file is "managed" so doing a config write to a keyval in that
+				// file would never work - it would get overwritten.
+				if c.cfgs[i].name == "z-50-fioctl.toml" {
+					return nil, fmt.Errorf("unable to override config-managed file: %s", c.cfgs[i].path)
+				}
+				return c.cfgs[i], nil
 			}
 		}
-		if found {
-			break
-		}
 	}
+	return nil, ErrNoWritableFound
```

Comment:
IIUC we only use this for cert rotation ATM, so at least one key is always present in at least one config file.

But, as this is a generic method, if no file contains a key - it does not mean that we cannot write that key to some file.
In fact, we may write to any file which is not `z-50-fioctl.toml` in this case.

Reply:
I think we need to be more explicit - also how do you determine the new file name? You have to guess or create one that sorts lexicographically after ever other file we might be parsing.

Reply:
I did not mean to create a new file.

What I meant is that given files as below, and trying to write `foo=bar`, we can write to either of those files.
```
sota1.toml:
  some=value

sota2.toml:
  other=value
```

Reply:
I think I'd rather change the name of the method to something like `UpdateKeys` or just making it a private function explicitly for key rotation. fioconfig shouldn't be *adding* configuration

Reply:
> change the name of the method ... making it a private function explicitly for key rotation

Yes, let's do it.
It is better to be explicit about what this function is about.

If we _later_ have a necessity to reuse this code, then we will _later_ decide how to do that.

Reply:
That said, if that would take more than 5 minutes of your time, I'm fine with a one-line comment here like e.g. `If the key is absent in all config files - we don't try to add it.`
I think that would be explicit enough.

Reply:
rebased on latest main and force pushed this change to filter into each commit of the series